meson.build

```diff
@@ -0,0 +1,819 @@
+            '-fno-vectorize',
+
+            # Use color in messages
+            '-fdiagnostics-show-option',
+            '-fcolor-diagnostics',
+            language:'cpp'
+        )
+        if cpp.version().version_compare('>=6.0')
+            add_project_arguments(
+                '-Wc++2a-extensions',
+                language:'cpp'
+            )
+        endif
+    endif
+    if is_windows
+        if cpp.get_id() == 'clang'
+            add_project_arguments(
+                '-Wno-global-constructors',
+                '-Wno-language-extension-token',
+                '-Wno-used-but-marked-unused',
+                '-Wno-shadow-field-in-constructor',
+                '-Wno-unused-member-function',
+                '-Wno-unused-template',
+                '-Wno-c++98-compat-pedantic',
+                '-Wno-used-but-marked-unused',
+                '-Wno-zero-as-null-pointer-constant',
+                language:'cpp',
+            )
+        endif
+        add_project_arguments(
+            '-Wno-cast-align',
+            '-Wno-double-promotion',
+            '-Wno-float-equal',
+            '-Wno-format-nonliteral',
+            '-Wno-shadow',
+            '-Wno-sign-conversion',
+            language:'cpp',
+        )
+    else
+
+        add_project_arguments(
+            '-fmath-errno',
+            # '-fno-exceptions', # handled by cpp_eh=none default option
+            language:'cpp'
+        )
+    endif
+
+    if get_option('sse2')
+        add_project_arguments(
+            '-msse2',
+            '-mfpmath=sse',
+            language:'cpp',
+        )
+    endif
+
+    if cpp.get_id() == 'gcc' or (cpp.get_id() == 'clang' and cpp.version().version_compare('>=11.0'))
+        add_project_arguments(
+            '-Wno-psabi',
+            language: 'cpp',
+        )
+    endif
+
+    if get_option('arm7')
+        add_project_arguments(
+            '-march=armv7-a',
+            '-mfpu=neon-vfpv4',
+            '-mfloat-abi=hard',
+            '-DHWY_HAVE_SCALAR_F16_TYPE=0',  # See #2625
+            '-DHWY_NEON_HAVE_F16C=0',
+            language: 'cpp',
+        )
+        if cpp.get_id() == 'gcc'
+            add_project_arguments(
+                '-mfp16-format=ieee',
+                language: 'cpp',
+            )
+        endif
+    endif
+
+    if have_riscv and get_option('rvv')
+        riscv_arg = ''
+        if riscv_xlen == '64'
+            riscv_arg = '-march=rv64gcv1p0'
+        elif riscv_xlen == '32'
+            riscv_arg = '-march=rv32gcv1p0'
+        endif
+
+        if riscv_arg != ''
+            add_project_arguments(riscv_arg, language: 'cpp')
+            add_project_link_arguments(riscv_arg, language: 'cpp')
+        endif
+
+        if cpp.get_id() == 'clang'
+            add_project_arguments('-menable-experimental-extensions', language: 'cpp')
+        endif
+    endif
+
+    if have_emscripten
+        add_project_arguments('-matomics', language:'cpp')
+    endif
```

Comment:
@WillAyd  I'd be curious about whether these sorts of arguments *should* be passed on up to projects using the highway dependency object, as they likely either effect how the included headers work, or they're already appropriate flags for these targets (in which case someone using the dependency object would likely also have to add these to their when compiling code targeting these architectures.

Reply:
Hmm I'm not sure - someone else probably has a better point of view on that. Looking at the CMake config though I believe these are not passed along transitively to projects that use highway, so maybe this is correct as is

Reply:
Ah I thought your question was more about whether these need to be used when linking against highway, to which I'm not sure. But if these are required for third party libraries, I think you should still do what you've done already with the export macros, i.e. include it in the `compile_args` argument of the `declare_dependency` call.